src/main.cpp

```diff
@@ -2497,14 +2497,20 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 }
 
 
-CConsensusContextInfo GetContextInfo(const CBlockIndex* pindex)
+CConsensusContextInfo GetContextInfo(const CBlockIndex* pindex, const Consensus::Params consensusParams)
 {
     assert(pindex);
     CConsensusContextInfo contextInfo;
     contextInfo.height = pindex->nHeight;
     contextInfo.medianTimePast = pindex->GetMedianTimePast();
     contextInfo.now = GetAdjustedTime();
     contextInfo.bestBlock = pindex->GetBlockHeader();
```

Comment:
Do you plan to move GetContextInfo() to libconsensus ? If not, please don't move consensus critical code out of consensus critical functions.